zap/src/main/java/org/parosproxy/paros/Constant.java

```diff
@@ -1271,6 +1277,51 @@ private static void upgradeFrom2_11_1(XMLConfiguration config) {
         config.setProperty("hud.enableTelemetry", null);
     }
 
+    static void upgradeFrom2_14_0(XMLConfiguration config, boolean save)
+            throws ConfigurationException {
+        List<HierarchicalConfiguration> existingTagScanners =
+                config.configurationsAt("pscans.autoTagScanners.scanner");
+
+        int index = existingTagScanners.size(); // Size is next index due to 0th start
+
+        String leadingAutoTagScannersKey = "pscans.autoTagScanners.scanner(";
+        String trailingNameKey = ").name";
+        String trailingTypeKey = ").type";
+        String trailingConfigKey = ").config";
+        String trailingResHeadRegex = ").resHeadRegex";
+        String trailingEnableKey = ").enabled";
+
+        config.setProperty(leadingAutoTagScannersKey + index + trailingNameKey, "json_extended");
```

Comment:
Will switch to add

Reply:
Why not check if it exists rather than adding duplicates? (Though should be fine with add, they will be dropped when parsing as duplicates are not allowed.)

Reply:
When would it add duplicates? Only when someone has a rule that happens to have the same name 🤷‍♂️

The upgrade should only run once per install. So it isn’t like potential rules are being added over and over.

Reply:
I guess we could collect all the description properties and check if the desc of the one being added is in the existing list? I dunno what else we could do?

Reply:
No need to collect, just get them all directly (i.e. `getList()`) and do a contains.